config/config_test.go

```diff
@@ -44,42 +43,26 @@ func (prefixValidator) Validate() error {
 
 func TestFromEnv(t *testing.T) {
 	subtests := []struct {
-		name    string
-		env     map[string]string
-		options EnvOptions
-		io      Validator
-		error   bool
+		name  string
+		opts  EnvOptions
+		io    Validator
+		error bool
 	}{
-		{"nil", nil, EnvOptions{}, nil, true},
-		{"nonptr", nil, EnvOptions{}, simpleValidator{}, true},
-		{"nilptr", nil, EnvOptions{}, (*simpleValidator)(nil), true},
-		{"defaulterr", nil, EnvOptions{}, new(nonStructValidator), true},
-		{"parseeerr", map[string]string{"FOO": "bar"}, EnvOptions{}, &simpleValidator{}, true},
-		{"invalid", map[string]string{"FOO": "23"}, EnvOptions{}, &simpleValidator{}, true},
-		{"simple", map[string]string{"FOO": "42"}, EnvOptions{}, &simpleValidator{42}, false},
-		{"default", nil, EnvOptions{}, &defaultValidator{42}, false},
-		{"override", map[string]string{"FOO": "23"}, EnvOptions{}, &defaultValidator{23}, false},
-		{"prefix", map[string]string{"PREFIX_FOO": "42"}, EnvOptions{Prefix: "PREFIX_"}, &simpleValidator{42}, false},
-		{"nested", map[string]string{"PREFIX_FOO": "42"}, EnvOptions{}, &prefixValidator{simpleValidator{42}}, false},
-	}
-
-	allEnv := make(map[string]struct{})
-	for _, st := range subtests {
-		for k := range st.env {
-			allEnv[k] = struct{}{}
-		}
+		{"nil", EnvOptions{}, nil, true},
+		{"nonptr", EnvOptions{}, simpleValidator{}, true},
+		{"nilptr", EnvOptions{}, (*simpleValidator)(nil), true},
+		{"defaulterr", EnvOptions{}, new(nonStructValidator), true},
+		{"parseeerr", EnvOptions{Environment: map[string]string{"FOO": "bar"}}, &simpleValidator{}, true},
+		{"invalid", EnvOptions{Environment: map[string]string{"FOO": "23"}}, &simpleValidator{}, true},
+		{"simple", EnvOptions{Environment: map[string]string{"FOO": "42"}}, &simpleValidator{42}, false},
+		{"default", EnvOptions{}, &defaultValidator{42}, false},
+		{"override", EnvOptions{Environment: map[string]string{"FOO": "23"}}, &defaultValidator{23}, false},
+		{"prefix", EnvOptions{Environment: map[string]string{"PREFIX_FOO": "42"}, Prefix: "PREFIX_"}, &simpleValidator{42}, false},
+		{"nested", EnvOptions{Environment: map[string]string{"PREFIX_FOO": "42"}}, &prefixValidator{simpleValidator{42}}, false},
```

Comment:
While on it, you don't have to rebase. I've hard-reset my branch to this ☝️ first commit.